src/modules/performance_collector/db.py

```diff
@@ -0,0 +1,168 @@
+import sqlite3
+from contextlib import contextmanager
+from typing import Optional
+
+from src import variables
+from src.modules.performance_collector.codec import ProposalDuty, SyncDuty, EpochDataCodec, AttDutyMisses
+from src.types import EpochNumber
+from src.utils.range import sequence
+
+
+class DutiesDB:
+    def __init__(self, path: str):
+        self._path = path
+        self._conn = sqlite3.connect(
+            self._path, check_same_thread=False, timeout=variables.PERFORMANCE_COLLECTOR_DB_CONNECTION_TIMEOUT
+        )
```

Comment:
Doesn't it make sense to have it in the `connection` context manager instead?

Reply:
I guess `check_same_thread=False` doesn't make sense here because the web server creates a new connection for every request anyway.